aquadoggo/src/materializer/tasks/schema.rs

```diff
@@ -1,9 +1,281 @@
+                related_schema_definitions.push(schema.id().clone())
+            } else {
+                continue;
+            }
+        } else {
+            // Abort if there are schema definitions in the store that don't match the schema
+            // definition schema.
+            Err(TaskError::Critical)?
+        }
+    }
+
+    Ok(related_schema_definitions)
+}
+
+#[cfg(test)]
+mod tests {
+    use log::debug;
+    use std::convert::TryFrom;
+
+    use p2panda_rs::document::DocumentViewId;
+    use p2panda_rs::entry::{sign_and_encode, Entry};
+    use p2panda_rs::identity::{Author, KeyPair};
+    use p2panda_rs::operation::{
+        Operation, OperationEncoded, OperationFields, OperationValue, PinnedRelationList,
+        VerifiedOperation,
+    };
+    use p2panda_rs::schema::{FieldType, SchemaId, SchemaVersion};
+    use p2panda_rs::storage_provider::traits::{OperationStore, StorageProvider};
+    use rstest::rstest;
+
+    use crate::context::Context;
+    use crate::db::stores::test_utils::{test_db, TestSqlStore};
+    use crate::graphql::client::{EntryArgsRequest, PublishEntryRequest};
+    use crate::materializer::tasks::reduce_task;
+    use crate::materializer::TaskInput;
+    use crate::schema::SchemaProvider;
+    use crate::Configuration;
+
+    use super::schema_task;
+
+    /// Insert a test schema definition and schema field definition and fire reduce tasks for both.
+    async fn insert_test_schema(
+        context: &Context,
+        key_pair: &KeyPair,
+    ) -> (DocumentViewId, DocumentViewId) {
+        let author = Author::try_from(key_pair.public_key().to_owned()).unwrap();
+
+        // Publish schema field definition
+        let args = EntryArgsRequest {
+            author: author.clone(),
+            document: None,
+        };
+        let args = context.store.get_entry_args(&args).await.unwrap();
+
+        let mut fields = OperationFields::new();
+        fields
+            .add("name", OperationValue::Text("message".to_string()))
+            .unwrap();
+        fields.add("type", FieldType::String.into()).unwrap();
+        let create_op = Operation::new_create(SchemaId::SchemaFieldDefinition(1), fields).unwrap();
+
+        let entry = Entry::new(
+            &args.log_id,
+            Some(&create_op),
+            args.skiplink.as_ref(),
+            args.backlink.as_ref(),
+            &args.seq_num,
+        )
+        .unwrap();
+        let entry_encoded = sign_and_encode(&entry, key_pair).unwrap();
+        let args = PublishEntryRequest {
+            entry_encoded: entry_encoded.clone(),
+            operation_encoded: OperationEncoded::try_from(&create_op).unwrap(),
+        };
+        let response = context.store.publish_entry(&args).await.unwrap();
+        let field_view_id: DocumentViewId = response.backlink.unwrap().into();
+
+        let verified_op =
+            VerifiedOperation::new(&author, &entry_encoded.clone().hash().into(), &create_op)
+                .unwrap();
```

Comment:
Same here as I commented above, you could insert the schema_field_definition through the `test_db` fixture then the following `schema_definition` using the `insert_entry_operation_and_view()` helper.

Reply:
You could also do it all through the fixture interface if you hard-code the `schema_field_definition` relation hash, maybe this is a little too funky though ;-p

Reply:
I wrote a reply in the other thread here https://github.com/p2panda/aquadoggo/pull/166#discussion_r921190910